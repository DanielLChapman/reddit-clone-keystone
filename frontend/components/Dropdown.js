import React, { useState } from 'react';
import PropTypes from 'prop-types'

function Dropdown(props) {
    /*
        let options = ['Hot', 'Best', 'New', 'Top', 'Rising'];
        const [dropValue, setDropValue] = useState('Hot');
        <Dropdown returnFunc={setDropValue} selected={dropValue} options={options} />    
    */
    return (
        <select value={props.selected} onChange={(e) => {
            props.returnFunc(e.target.value);
        }}>
            {
                props.options && props.options.map((x, i) => (
                    <option key={i} value={x}>{x}</option>
                ))
            }
        </select>
    );
}

Dropdown.propTypes = {
    returnFunc: PropTypes.func.isRequired,
    options: PropTypes.array.isRequired,
    selected: PropTypes.string.isRequired,
};
  

export default Dropdown;