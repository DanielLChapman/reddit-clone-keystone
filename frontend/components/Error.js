import React from 'react';

function Error(props) {
    return (
        <span>
            {props.error}
        </span>
    );
}

export default Error;