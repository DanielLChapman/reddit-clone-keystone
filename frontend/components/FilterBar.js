import React, { Component } from 'react';
import { FaHotjar, FaBahai, FaGopuram, FaChartLine} from 'react-icons/fa';



class FilterBar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            li1: true,
            li2: false,
            li3: false,
            li4: false
        }
    }

    toggleClass = (num) => {
        let state = {
            li1: false,
            li2: false,
            li3: false,
            li4: false
        };
        state['li'+num] = true;
        this.setState({...state});
    }

    render() {
        let style = {
            display: "none"
        };
        if (this.state.li3) {
            style= {
                display: "inline-block"
            };
        }
        return (
            <section className="filter-bar"> 
                <h5>Popular Posts</h5>
                <ul>
                    <li className={this.state.li1 ? 'active' : null} onClick={() => this.toggleClass(1)}><FaHotjar /> Hot</li>
                    <li className={this.state.li2 ? 'active' : null} onClick={() => this.toggleClass(2)}><FaBahai /> New</li>
                    <li className={this.state.li3 ? 'active' : null} onClick={() => this.toggleClass(3)}><FaGopuram /> Top</li>
                    <li className="time-selector-for-filter-bar" style={style}>Today</li>
                    <li className={this.state.li4 ? 'active' : null} onClick={() => this.toggleClass(4)}><FaChartLine /> Rising</li>
                </ul>
            </section>
        );
    }
}

export default FilterBar;