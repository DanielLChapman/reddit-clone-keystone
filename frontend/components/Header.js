import Link from 'next/link';
import Nav from './Nav';


export default function Header() {
  return (
    <>
      <div className="bar">
        <span>
          <Link href="/">App name</Link>
        </span>
        <Nav />
      </div>
    </>
  );
}
