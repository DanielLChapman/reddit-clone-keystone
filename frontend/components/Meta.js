import Head from 'next/head';

const Meta = () => (
    <Head>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <meta charSet="utf-8" />
        <link rel="shortcut icon" href="/favicon.png" />
        <link rel="stylesheet" type="text/css" href="/nprogress.css" />
        <link rel="stylesheet" type="text/css" href="/main.css" />
        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css"></link>
        <title>Reddit Clone</title>
        <script async src="//s.imgur.com/min/embed.js" charSet="utf-8"></script>
    </Head>

)

export default Meta;