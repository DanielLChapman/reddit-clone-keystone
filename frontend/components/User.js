import { gql, useQuery } from '@apollo/client';

export const CURRENT_USER_QUERY = gql`
  query {
    authenticatedItem {
      ... on User {
        id
        username
        email
        name
        posts {
          votes {
            vflag
          }
        }
        comments {
          votes {
            vflag
          }
        }
        moderating {
          id
          slug
          name
        }
        owner {
          id
          slug
          name
        }
        subreddits {
          id
          slug
          name
        }
        postvotes {
          post {
            title
            id
          }
          id
          vflag
        }
        commentvotes {
          comment {
            id
          }
          id
          vflag
        }
    }
  }
  }
`;

export function useUser() {

  const { data } = useQuery(CURRENT_USER_QUERY);
  let test = {...data};

  
  if (test.authenticatedItem) {
    let x = Object.assign({}, test.authenticatedItem);
    //count up post votes
    let postVotes = 0;
    let commentVotes = 0;
    
    x.posts.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? postVotes += 1 : postVotes -= 1;
      })
    });
    //count up comment votes
    x.comments.forEach((q) => {
      q.votes.forEach((y) => {
        y.vflag === 'Upvote' ? commentVotes += 1 : commentVotes -= 1;
      })
    });

    x.postVotes = postVotes;
    x.commentVotes = commentVotes;

    return x;
  } else {
    return data?.authenticatedItem;
  }

  
}
