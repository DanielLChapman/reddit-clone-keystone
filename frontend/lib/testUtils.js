import casual from 'casual';

  
// seed it so we get consistent results
casual.seed(777);

const fakeUser = (overrides) => (
    {
    __typename: 'User',
    id: '4234',
    name: casual.name,
    username: casual.name,
    owner: [],
    email: casual.email,
    permissions: ['ADMIN'],
    moderating: [],
    subreddits: [],
    posts: [],
    postvotes: [],
    firstSignin: false,
    comments: [],
    commentvotes:[],
    ...overrides,
});

const fakeSubreddit = (overrides) => (
    {
        __typename: 'Subreddit',
        id: casual.uuid,
        name: casual.string,
        title: casual.string,
        slug: casual.string,
        sidebar: casual.string,
        rules: casual.string,
        description: casual.string,
        status: 'PUBLIC',
        ownder: fakeUser(),
        moderators: [],
        subscriber: [],
        posts: [],
        createdAt: Date.now(),
    }
);

const fakeLinkPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'www.google.com',
    type: 'Link',
    comments: [],
    removed: 'False',
    id: casual.uuid,


});

const fakeMediaPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: 'https://cdn.fileinfo.com/img/ss/sm/png_79.jpg',
    type: 'Link',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeTextPost = (overrides) => (
    {
    __typename: 'Post',
    title: casual.sentence,
    content: casual.sentence,
    user: fakeUser(),
    subreddit: fakeSubreddit(),
    createdAt: Date.now(),
    votes: [],
    post_slug: casual.string,
    link: '',
    type: 'Text',
    comments: [],
    id: casual.uuid,
    removed: 'False'

});

const fakeComment = (overrides) => (
    {
    __typename: 'Comment',
    content: casual.sentence,
    user: fakeUser(),
    post: fakeTextPost(),
    createdAt: Date.now(),
    votes: [],
    parent: null,
    id: casual.uuid,
    children: [],

});


// Fake LocalStorage
class LocalStorageMock {
    constructor() {
      this.store = {};
    }
  
    clear() {
      this.store = {};
    }
  
    getItem(key) {
      return this.store[key] || null;
    }
  
    setItem(key, value) {
      this.store[key] = value.toString();
    }
  
    removeItem(key) {
      delete this.store[key];
    }
  }



export {

    LocalStorageMock,
    fakeLinkPost,
    fakeComment,
    fakeMediaPost,
    fakeSubreddit,
    fakeTextPost,
    fakeUser,

  };