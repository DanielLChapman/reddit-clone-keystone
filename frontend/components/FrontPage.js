import Link from 'next/link';
import React from 'react';
import { useUser } from './User';

function FrontPage(props) {
    const user = useUser();
    return (
        <div>
            {user && (
                <Link href="/subreddits/create">Create Your Own Subreddit</Link>
            )}
            
        </div>
    );
}

export default FrontPage;