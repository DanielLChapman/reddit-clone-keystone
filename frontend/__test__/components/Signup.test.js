import { render, screen } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import userEvent from '@testing-library/user-event';
import wait from 'waait';
import Signup, { SIGNUP_MUTATION } from '../../components/SignUp';
import { CURRENT_USER_QUERY } from '../../components/User';
import { fakeUser } from '../../lib/testUtils';
import { act } from 'react-dom/test-utils';

const me = fakeUser();

const password = 'wesswess1';


const mocks = [
  // Mutation Mock
  {
    request: {
      query: SIGNUP_MUTATION,
      variables: {
        name: me.name,
        email: me.email,
        username: me.username,
        case_username: me.username.toLowerCase(),
        password,
      },
    },
    result: {
      data: {
        createUser: {
          __typename: 'User',
          id: 'abc123',
          email: me.email,
          name: me.name,
        },
      },
    },
  },
  // Current user mock
  // {
  //   request: { query: CURRENT_USER_QUERY },
  //   result: { data: { authenticatedItem: me } },
  // },
];

describe('<SignUp/>', () => {
    
  it('render and matches snapshot', () => {
    const { container } = render(
      <MockedProvider>
        <Signup />
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('calls the mutation properly', async () => {
    const { container, debug } = render(
      <MockedProvider mocks={mocks}>
        <Signup />
      </MockedProvider>
    );
    // Type into the boxes
    await userEvent.type(screen.getByPlaceholderText('Name'), me.name);
    await userEvent.type(screen.getByPlaceholderText('Username'), me.username);
    await userEvent.type(screen.getByPlaceholderText(/email/i), me.email);
    await userEvent.type(screen.getByPlaceholderText(/password/i), password);
    // Click the submit
    await userEvent.click(screen.getByText('Sign Up'));
    await screen.findByTestId('success-signup-message');

  });

  it('calls the mutation with errors', async () => {
    const { container, debug } = render(
        <MockedProvider mocks={mocks}>
          <Signup />
        </MockedProvider>
      );
      // Type into the boxes
      await userEvent.type(screen.getByPlaceholderText('Name'), me.name);
      await userEvent.type(screen.getByPlaceholderText('Username'), me.username);
      await userEvent.type(screen.getByPlaceholderText(/email/i), me.email);
      await userEvent.type(screen.getByPlaceholderText(/password/i), 'test');

      await userEvent.click(screen.getByText('Sign Up'));
      await screen.findByText(
        'Password required to be a minimum of 8 characters'
      );
  })

});