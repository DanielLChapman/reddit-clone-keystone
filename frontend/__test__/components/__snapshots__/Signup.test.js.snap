// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp/> render and matches snapshot 1`] = `
<div>
  <form
    class="Form-sc-1tx35ym-0 etUUQp"
    method="POST"
  >
    <h2>
      Sign Up For An Account
    </h2>
    <fieldset>
      <label
        for="name"
      >
        Username
        <input
          autocomplete="username"
          name="username"
          placeholder="Username"
          type="text"
          value=""
        />
      </label>
      <label
        for="name"
      >
        Name
        <input
          autocomplete="name"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </label>
      <label
        for="email"
      >
        Email
        <input
          autocomplete="email"
          name="email"
          placeholder="Your Email Address"
          type="email"
          value=""
        />
      </label>
      <label
        for="password"
      >
        Password
        <input
          autocomplete="password"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </fieldset>
  </form>
</div>
`;
